probe-rs/src/flashing/flasher.rs

```diff
@@ -805,12 +805,29 @@ impl<'p> ActiveFlasher<'p, Program> {
         );
 
         // TODO: Prevent security settings from locking the device.
-
         // Transfer the buffer bytes to RAM.
+        
+        let t1 = std::time::Instant::now();
+        let words: Vec<u32> = bytes
+            .chunks_exact(core::mem::size_of::<u32>())
+            .map(|a| u32::from_le_bytes([a[0], a[1], a[2], a[3]]))
+            .collect();
+
         self.core
-            .write_8(algo.page_buffers[buffer_number], bytes)
+            .write_32(algo.page_buffers[buffer_number], &words)
             .map_err(FlashError::Core)?;
 
+        let rem = bytes.len() - (words.len() * core::mem::size_of::<u32>());
```

Comment:
If we enforce that page size *must* be a multiple of 4, we can remove this entirely